ui/org.eclipse.pde.core/src/org/eclipse/pde/core/plugin/PluginRegistry.java

```diff
@@ -45,15 +46,24 @@
  */
 public class PluginRegistry {
 
+	private PluginRegistry() { // static use only
+	}
+
 	/**
 	 * Filter used when searching for plug-in models.
 	 * <p>
 	 * Clients may subclass this class to implement custom filters.
 	 * </p>
-	 * @see PluginRegistry#findModel(String, String, int, PluginFilter)
-	 * @see PluginRegistry#findModel(String, VersionRange, PluginFilter)
+	 *
+	 * @see PluginRegistry#findModels(String, String, VersionMatchRule)
+	 * @see PluginRegistry#findModels(String, VersionRange)
 	 * @since 3.6
+	 * @deprecated Instead use {@link Predicate} and filter the stream returned
+	 *             by
+	 *             {@link PluginRegistry#findModels(String, String, VersionMatchRule)}
+	 *             or {@link PluginRegistry#findModels(String, VersionRange)}
 	 */
+	@Deprecated(forRemoval = true, since = "3.19 (removal in 2026-09 or later)")
```

Comment:
Btw. what do you think about mentioning the earliest removal date here?
I think it would help consumers to understand better when the actual removal will happen earliest.